### ISSUE TITLE
Help

### ISSUE THREAD

**Issue by Rantoledo** [fastify/help#1070]
## 💬  Question here 

How does `@fastify/multipart` work?
I'm trying to upload a file approximately 20-50 mib size.
I'm planning on using multipart as a content type, so I used `@fastify/multipart`.
But, I have `nginx` reverse proxy, so I'm limited to 1mb to the request size.
My question is does multipart send the data in chunks? does it do that for me? my testing saying it doesn't, but just wanted to ask to make sure.

- *node version*: 22
- *fastify version*: 5.1.0
- *os*:  Linux

**Comment by climba03003** [comment]
> My question is does multipart send the data in chunks?

Yes, by the nature of stream it is always send by chunk. The only difference is the chunk size.

> does it do that for me?

By default, using the Stream API should be process by chunk.
However, `@fastify/multipart` offers the ability to buffer in memory. So, it may gives some negative image.

> my testing saying it doesn't, but just wanted to ask to make sure.

You should probably post the code you test. There is no way to ensure your implementation is doing things right.

**Comment by Rantoledo** [comment]
@climba03003 
So, you're saying that each chunk is sent in a separate request, and the only reason my backend receives a single request with the entire file is because of `nginx`?

**Comment by climba03003** [comment]
Possibly, yes.
